core/src/test/java/io/substrait/type/proto/JoinRoundtripTest.java

```diff
@@ -57,4 +58,15 @@ void hashJoin() {
             .build();
     verifyRoundTrip(relWithoutKeys);
   }
+
+  @Test
+  void nestedLoopJoin() {
+    Rel rel =
+        NestedLoopJoin.builder()
+            .from(
+                b.nestedLoopJoin(
+                    __ -> b.bool(true), NestedLoopJoin.JoinType.INNER, leftTable, rightTable))
```

Comment:
The test in ExtensionRoundtripTest.java use a bool true condition as well. Could you use a more interesting condition here like a key equality comparison between the two tables (ie. `leftTable.a = rightTable.f`) so that we can have a test that verifies that the condition expressions are being converted correctly.

Reply:
Good point! Will update.